### PR TITLE
Do not emit alerts when the counter overflows

### PR DESCRIPTION
**Pull request by cyones**

### CODE REVIEW

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py [hunk 1]
```diff
@@ -165,16 +165,17 @@ def process_packet(packet, policy):
...
-                if not (packet.f_count == device_session.up_link_counter and last_uplink_mic[device_session.id] == packet.mic): 
-                    emit_alert("LAF-007", packet, device=device, device_session=device_session, gateway=gateway,
-                                counter=device_session.up_link_counter,
-                                new_counter=packet.f_count,
-                                prev_packet_id=device_session.last_packet_id)
+        elif (
+            policy.is_enabled("LAF-007") and
+            device_session.id in last_uplink_mic and
+            packet.f_count > 5 and device_session.up_link_counter < 65530 and
+            packet.f_count < device_session.up_link_counter and
```
Comment (matiassequeira) [inline analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py]: Is it considered the case when we receive the same counter but different MIC?

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py [hunk 2]
```diff
@@ -165,16 +165,17 @@ def process_packet(packet, policy):
 
                     device_session.reset_counter += 1
         
-        elif packet.f_count <= device_session.up_link_counter:
-
-            # Make sure we have processed at least one packet for this device in this run before firing the alarm
```
Comment (matiassequeira) [inline analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py:170]: Don't leave comments and explanations out

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py [hunk 3]
```diff
@@ -165,16 +165,17 @@ def process_packet(packet, policy):
...
-                # Skip if received the same counter as previous packet and mics are equal
-                if not (packet.f_count == device_session.up_link_counter and last_uplink_mic[device_session.id] == packet.mic): 
-                    emit_alert("LAF-007", packet, device=device, device_session=device_session, gateway=gateway,
-                                counter=device_session.up_link_counter,
-                                new_counter=packet.f_count,
-                                prev_packet_id=device_session.last_packet_id)
+        elif (
+            policy.is_enabled("LAF-007") and
+            device_session.id in last_uplink_mic and
+            packet.f_count > 5 and device_session.up_link_counter < 65530 and
```
Comment (matiassequeira) [inline analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py]: If we receive packet.f_count==3 and device_session.up_link_counter==4, the alarm wouldn't be triggered? I'd split this condition

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py (hunk 1 above)
Comment (cyones) [reply analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py]: Fixed

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py (hunk 2 above)
Comment (cyones) [reply analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py:170]: Comments added

File: analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py (hunk 3 above)
Comment (cyones) [reply analyzers/rolaguard_base_analyzer/BaseAnalyzerMain.py]: Fixed

**Review by matiassequeira — APPROVED** [review]

**Pull request merged**